src/main/java/pl/javastart/task/CustomList.java

```diff
@@ -1,6 +1,69 @@
 package pl.javastart.task;
 
+import java.util.Arrays;
+
 // Uzupełnij klasę. Możesz ją dowolnie modyfikować.
 // Celem jest, żeby przechodziły testy w src/test/java/[...]/CustomListTest
-public class CustomList {
+public class CustomList<T> {
+    private T[] elements;
+    private int size = 0;
+
+    private void checkOrEnoughCapacity(int capacity) {
+        if (elements == null) {
+            elements = (T[]) new Object[1];
+        } else {
+            if (capacity > elements.length) {
+                int newSize = size + 1;
+                elements = Arrays.copyOf(elements, newSize);
+            }
+        }
```

Comment:
Lista tablicowa z założenia działa w taki sposób, że pod spodem ma pustą tablicę z rozmiarem "na zapas", którą uzupełniamy i w razie potrzeby powiększamy. Przykładowo lista, w której będzie tylko jeden element, w rzeczywistości pod spodem może wykorzystywać tablicę, w której jest np. 10 miejsc, tzn. `["a", null, null, null, null, ...]`

Lepiej jest zwiększać tablicę o jakąś większą wartość niż 1 i na początku warto stworzyć też tablicę, w której jest od razu np. 10 wolnych miejsc.
Jeżeli zwiększamy tablicę tylko o 1, to wyobraź sobie dodawanie np. 10_000 elementów - musielibyśmy 10_000 razy kopiować tablicę do nowej tablicy co jest bardzo nieefektywne.
Raczej jest też mało prawdopodobne, że będziemy tworzyli listę, w której będzie tylko 1 element.
W najprostszym przypadku możesz sobie założyć np. wyjściowy rozmiar jako 10, a później powiększać tablicę dwukrotnie, gdy jest taka potrzeba. 
(w rzeczywistości takie powiększanie x2 też nie jest idealne, bo jeżeli będziemy mieli np. bardzo dużą tablicę, w której już np. 100_000 obiektów, to nie chcemy jej powiększać o 100_000, tylko wystarczy np. dodatkowych 1000 miejsc. Na nasze potrzeby to jednak wystarczy)